pkg/ccl/changefeedccl/kafkaauth/kafkaauth.go

```diff
@@ -0,0 +1,163 @@
+package kafkaauth
+
+import (
+	"context"
+
+	"github.com/IBM/sarama"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/errors"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+type saslMechanismBuilder interface {
+	name() string
+	validateParams(u *changefeedbase.SinkURL) error
+	build(u *changefeedbase.SinkURL) (saslMechanism, error)
+}
+
+type saslMechanism interface {
+	ApplySarama(ctx context.Context, cfg *sarama.Config) error
+	KgoOpts(ctx context.Context) ([]kgo.Opt, error)
+}
+
+type saslMechanismRegistry map[string]saslMechanismBuilder
+
+// Registry is the global registry of SASL Mechanisms.
+var Registry saslMechanismRegistry = make(map[string]saslMechanismBuilder)
```

Comment:
Instead of having clients call `kafkaauth.Registry.Pick`, what if instead you exported a `Pick` function directly. This would also prevent external changes to the map.